src/app/games/trexo/tests/three-d-iso-square.component.spec.ts

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable max-lines-per-function */
+import { DebugElement } from '@angular/core';
+import { fakeAsync } from '@angular/core/testing';
+import { Coord } from 'src/app/jscaip/Coord';
+import { CoordXYZ } from 'src/app/jscaip/CoordXYZ';
+import { SimpleComponentTestUtils } from 'src/app/utils/tests/TestUtils.spec';
+import { ThreeDIsoSquareComponent } from '../three-d-iso-square.component';
+import { TrexoComponent } from '../trexo.component';
+import { TrexoMove } from '../TrexoMove';
+
+describe('ThreeDIsoSquareComponent', () => {
+
+    let testUtils: SimpleComponentTestUtils<ThreeDIsoSquareComponent>;
+
+    let component: ThreeDIsoSquareComponent;
+
+    beforeEach(fakeAsync(async() => {
+        testUtils = await SimpleComponentTestUtils.create(ThreeDIsoSquareComponent);
+        component = testUtils.getComponent();
+        component.coord = new CoordXYZ(1, 1, 0);
+        component.mode = TrexoComponent.modeMap['3D'];
+    }));
+
+    it('should create', fakeAsync(async() => {
+        // wait for the chat to be initialized (without it, ngOnInit will not be called)
+        testUtils.detectChanges();
+        expect(component).toBeTruthy();
+    }));
+
+    it(`should display rhombus as closed when missing move`, () => {
+        // Given a component on which no move is given
+
+        // When displaying it
+        testUtils.detectChanges();
+
+        // Then the component shoud have the 5 different potential piece (hence, a fully closed rhombus-prism)
+        testUtils.expectElementToExist('#tile_1_1_0');
+        testUtils.expectElementToExist('#right_up_1_1_0');
+        testUtils.expectElementToExist('#down_right_1_1_0');
+        testUtils.expectElementToExist('#right_down_1_1_0');
+        testUtils.expectElementToExist('#down_left_1_1_0');
+    });
+
+    it(`should display rhombus as open to the right when lefter part of move`, () => {
```

Comment:
"lefter part of move" waatt
when displays as the lef part of a move?

Reply:
idem les suivints

Reply:
ainsi lo ?

Reply:
why not: "when it is the left part of a move" ? (lefter ça n'existe po!)